### ISSUE TITLE
`U` should be mapped to "Undefine Current Function"

### ISSUE THREAD

**Issue by plafosse** [Vector35/binaryninja-api#3392]
'U' Should be mapped to "Undefine Current Function"

**Comment by plafosse** [comment]
This is slightly more involved than initial appearance as "Undefine Symbol" is already mapped to 'u' so instead 'Undefine Symbol' should be expaneded to also be able to undefine functions depending on context.

**Comment by plafosse** [comment]
Yeah the fear is that we'd be screwing up someones workflow. Perhaps thats fear is unfounded. I personally have never used "U" to undefine a symbol either.

**Comment by galenbwill** [comment]
While i originally remapped U to undefine function, there was That One Time when i hit it accidentally and it really screwed me up... so I remapped it to Ctrl-Shift-U.... and never looked back.

That being said, I am strongly in support of a context-sensitive one-action-to-Undefine-them-all, because I have to have a separate mapping for Undefine Variable, and I always forget what it is.

Also, I can't even figure out in what context Undefine Symbol gets enabled.

**Comment by FirehaK** [comment]
I actually mapped "U" to be undefine symbol because it is what I was used to coming from IDA. Undefine function got remapped to CTRL + ALT + U. When I press a keyboard shortcut, I don't expect an action to take place on the entire function if I didn't select it's start address, just where my selection is. To me, undefining just the thing I have selected is what makes sense.

https://www.hex-rays.com/products/ida/support/freefiles/IDA_Pro_Shortcuts.pdf
